pkg/api/types.go

```diff
@@ -123,8 +123,8 @@ type OrchestratorProfile struct {
 // KubernetesConfig contains the Kubernetes config structure, containing
 // Kubernetes specific configuration
 type KubernetesConfig struct {
-	KubernetesHyperkubeSpec string `json:"kubernetesHyperkubeSpec,omitempty"`
-	KubectlVersion          string `json:"kubectlVersion,omitempty"`
+	KubernetesImageBase string `json:"kubernetesImageBase,omitempty"`
+	KubectlVersion      string `json:"kubectlVersion,omitempty"`
```

Comment:
KubectlVersion is also used to concat the kubectl download url. Personally I prefer keep the kubectlversion because we can simplify the logic in customData. Also what if the image url is: hyperkube.amd64:latest? In that case we can never determine the version of kubectl.